src/target/riscv/riscv.c

```diff
@@ -4822,6 +4822,19 @@ static int riscv_set_or_write_register(struct target *target,
 
 	keep_alive();
 
+	if (rid == GDB_REGNO_PC) {
+		LOG_TARGET_DEBUG(target, "writing PC to DPC: 0x%" PRIx64, value);
```

Comment:
Can you remove this debug print? That would be the other half of #915.

Reply:
1. This bug was encountered when I used riscv-tests to test FPGA：
https://github.com/riscv-software-src/riscv-tests/blob/d4eaa5bd6674b51d3b9b24913713c4638e99cdd9/debug/gdbserver.py#L1591-L1601
When rid is priv, the cacheable read/write function is not used to handle dcsr, which may result in incorrect dcsr from cache.
2. I looked at several failed automated build operations, Check Code Style doesn't seem to be the cause for me, and I've fixed the other errors.
3. I will remove the debug print later and resubmit it.

Reply:
Thanks for explaining.